### DISCUSSION TITLE
`Shift-Enter` to open a program in run mode, do not close the terminal after running it

### DISCUSSION THREAD

**Discussion by c02y** [davatorium/rofi#1560] [Q&A] [answered]
When I try to open a program in bin from the list in rofi with terminal using `Shift-Enter`, some programs such as `cal` will not keep the terminal open, it closes the terminal immediately.

This kind of program usually just prints the result and returns to the new shell prompt, programs such as `top` or `bc` that don't return to the new shell prompt will be OK, the terminal for them will still be open.

So is there anything I can do to make the terminal keep opening no matter what program I try to run using `Shift-Return`?

BTW: I tried kitty, alacritty and gnome-terminal.

**Comment by DaveDavenport** [comment, marked as answer]
you can change how this behaves in the config by modifying the command rofi uses to run the application.
Just modify this to run application + something that keeps terminal open.

**Reply by c02y** [reply]
I'm sorry, I don't get it, could you please provide an example how to do that?

**Reply by DaveDavenport** [reply]
Modify the command that rofi executes when opening an application in terminal:

```markdown
      -run-shell-command cmd

       Set command to execute when running an application in a shell.  See PATTERN.

       Default: {terminal} -e {cmd}

```

**Reply by c02y** [reply]
Yeah, thank you, just found that kitty and alacritty have `--hold` option, passing something like `-run-shell-command 'kitty --hold -e {cmd}'` to rofi command line will work.

--------------------------------
And I was reading rofi man page, there is a line:
```
 • {terminal}: the configured terminal (see -terminal-emulator)
```

But I failed to find `-terminal-emulator` in `man rofi` or `rofi --help`, maybe it is outdated? I'm using `rofi 1.7.2`.

**Reply by DaveDavenport** [reply]
thx, that is a typo. its `-terminal`.  I will fix that.
